src/App.tsx

```
@@ -1,27 +1,46 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
+import { sortItems } from './Utils/RenderItems';
+import { GoodsType } from './types/GoodsType';
 // or
 // import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [items, setItems] = useState<Good[]>([]);
+  const [type, setType] = useState<GoodsType>('');
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  useEffect(() => {
+    const promise = sortItems(type);
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    if (promise) {
+      promise.then(setItems);
+    }
+  }, [type]);
```

This violates checklist item #3: 'Error handling must be implemented for server requests (e.g., try/catch or promise catch, and expose/reflect error state)'.

Currently, if the promise from `sortItems` rejects (e.g., network error), the error is not handled or displayed. You should add error handling logic to catch and display errors to the user.